### PR TITLE
Basic foodtruck service

### PR DESCRIPTION
**Pull request by mtomwing**

### CODE REVIEW

**Comment by bobcatfish** [comment]
DECREASED COVERAGE MICHAEL HOW COULD YOU

File: catinabox/services/util.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import functools
+import json
+
+import nameko.web.handlers
+
+
+def http(*args, **kwargs):
```
Comment (bobcatfish) [inline catinabox/services/util.py]: wut

can has comments/docstring?

File: catinabox/services/food_truck_service.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from .util import http
```
Comment (bobcatfish) [inline catinabox/services/food_truck_service.py]: why does food truck need special decorator?

**Comment by bobcatfish** [comment]
Seems reasonable, curious about the `htttp` decorator

File: catinabox/services/food_truck_service.py (hunk 2 above)
Comment (mtomwing) [inline catinabox/services/food_truck_service.py]: Was going to bring it over to the cattery too. It's just some magic to automatically `json.dumps` the return of an endpoint.

e.g.

``` python
return 200, {'foo': 'bar'} ====> return 200, json.dumps({'foo': 'bar'})
```

Comment (bobcatfish) [inline catinabox/services/food_truck_service.py]: What do you think about calling it something like `http_json`?

Comment (mtomwing) [inline catinabox/services/food_truck_service.py]: Could do. I'll rename it.

File: catinabox/food_truck.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
+class FoodDoesNotExistError(Exception):
+    """The requested food item does not exist in the inventory."""
+
+    def __init__(self, food):
+        super(FoodDoesNotExistError, self).__init__(
+            'Food {!r} was not found in the inventory'.format(food))
```
Comment (bobcatfish) [inline catinabox/food_truck.py:6]: Note to @bobcatfish I need to update the other exceptions to be in this format

File: catinabox/food_truck.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    @property
+    def inventory(self):
+        return self._inventory
+
+    def stock(self, food, quantity):
+        """Add a new food item to the inventory.
+
+        Args:
+            food - name of the food item (string)
+            quantity - amount of food to stock initially (integer)
```
Comment (bobcatfish) [inline catinabox/food_truck.py:44]: I prefer the sphinx style docstrings that pycharm uses by default

File: catinabox/food_truck.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+    def __init__(self):
+        self._inventory = {}
+
+    @property
+    def inventory(self):
+        return self._inventory
+
+    def stock(self, food, quantity):
+        """Add a new food item to the inventory.
```
Comment (bobcatfish) [inline catinabox/food_truck.py:40]: if we keep `stock` and `restock` can the docstring explain that `stock` can only be called once?

File: catinabox/food_truck.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    def restock(self, food, quantity):
+        """Update the quantity of food in the inventory.
+
+        Args:
+            food - name of the food item (string)
+            quantity - amount of food to restock (integer)
+
+        """
+        if quantity < 0:
+            raise FoodQuantityCannotBeNegativeError(food, quantity)
```
Comment (bobcatfish) [inline catinabox/food_truck.py:64]: it can be 0 tho?

**Comment by bobcatfish** [comment]
Okay so still to do:
- @bobcatfish - Update exceptions in the rest of the code base
- @mtomwing - Update docstrings to be in sphinx format
- @mtomwing - Update cattery to use food truck

**Pull request merged**